src/main/java/hudson/remoting/Engine.java

```diff
@@ -678,6 +643,23 @@ public void afterChannel(@Nonnull JnlpConnectionState event) {
         }
     }
 
+    private JnlpEndpointResolver createEndpointResolver(List<String> jenkinsUrls) {
+        JnlpEndpointResolver resolver;
+        if (instanceIdentity == null) {
+            SSLSocketFactory sslSocketFactory = null;
+            try {
+                sslSocketFactory = getSSLSocketFactory();
+            } catch (Exception e) {
+                events.error(e);
```

Comment:
Should this perhaps return `null` on error or something? Otherwise, the following `resolver = ...` line could be partially initialized which doesn't sound right to me. Or should it use the default configurator like in line 658?

Reply:
This particular logic hasn't changed in this PR. It's just been moved around a little. It could probably be changed to be more robust but there aren't any complaints against it so it's probably not worth the bother.

Basically, if it fails to create the custom SSL socket factory it passes null and the following steps use the default one. Not necessarily a bad choice. The flow is a little odd, though, because most likely it has already done this same operation earlier. And there isn't much of a real reason why it shouldn't be able to create the socket factory.